File: 프로그래머스/2/87390. n＾2 배열 자르기/n＾2 배열 자르기.java
class Solution {
    public int[] solution(int n, long left, long right) {
       int[] answer = new int[(int) (right - left + 1)];

        int idx = 0;
        for (long i = left; i <= right; i++) {
            long row = i / n;
            long col = i % n;

            if (col <= row) {
                answer[idx++] = (int) row + 1;
            } else {
                answer[idx++] = (int) col + 1;
            }
        }

        return answer;
    }
}

File: 프로그래머스/2/87390. n＾2 배열 자르기/n＾2 배열 자르기.cpp
#include <string>
#include <vector>

using namespace std;

pair<int, int> GetIndex(int n, long long idx)
{
    return make_pair(idx / n, idx % n);
}

vector<int> solution(int n, long long left, long long right) {
    vector<int> answer;
    
    for(long long i = left; i <= right; ++i)
    {
        pair<int, int> pos = GetIndex(n, i);
        
        int num;
        if (pos.second > pos.first)
        {
            num = pos.second + 1;
        }
        else
        {
            num = pos.first + 1;
        }
        
        answer.emplace_back(num);
    }
    
    return answer;
}